engine/src/main/java/org/hibernate/search/query/engine/impl/EntityInfoImpl.java

```diff
@@ -77,12 +83,12 @@ public Serializable getId() {
 			 * Only throw an exception here, not in the constructor,
 			 * because in some cases we don't need the identifier at all (e.g. projections).
 			 */
-			if ( clazz == null ) {
+			if ( type == null ) {
 				//FIXME this is horrible, but a temporary refactoring as we move to IndexedTypeIdentifier
 				throw log.incompleteEntityInfo( null, null );
 			}
 			else {
-				throw log.incompleteEntityInfo( new PojoIndexedTypeIdentifier( clazz ), id );
+				throw log.incompleteEntityInfo( type, id );
```

Comment:
ah, yes. This was actually required, but in intermediate stages which are long gone.